fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java

```diff
@@ -510,6 +510,8 @@ private void eliminateFromGroupAndMoveToTargetGroup(Group fromGroup, Group targe
                     "TargetGroup should be ancestors of fromGroup, but fromGroup is root. Maybe a bug");
         }
 
+        moveParentExpressionsReference(fromGroup, targetGroup);
```

Comment:
In the context, `fromGroup` is the group of `LogicalRelation`, and the `targetGroup` is the group of `LogicalRowPolicy`, `eliminateFromGroupAndMoveToTargetGroup` will move the logicalExpression(LogicalRowPolicy) to the targetGroup. And the origin logicalExpression in the targetGroup will be recycle until the `LogicalRelation`.

So I don't think we should `moveParentExpressionsReference(fromGroup, targetGroup)`